internal/controller/gcpupgrade/inputs.go

```diff
@@ -0,0 +1,257 @@
+			input.Title = "Zone"
+			input.Default = "us-central1-c"
+			input.HelpText = "Zone is the Google zone where the (to be installed) Flightcrew virtual machine instance will be located."
+			maybeSetValue(keyZone)
+
+		case keyTowerVersion:
+			input = wrapinput.NewFreeForm()
+			input.Freeform.Placeholder = "stable"
+			input.Freeform.CharLimit = 32
+			input.Title = "Tower Version"
+			input.HelpText = "Tower Version is the version of the Tower image that will be installed. (recommended: `stable`)"
+			maybeSetValue(keyTowerVersion)
+
+		}
+
+		input.Blur()
+		ctl.inputs[key] = &input
+	}
+	return ctl
+}
+
+func (ctl InputsController) GetAllInputs() []*wrapinput.Model {
+	res := make([]*wrapinput.Model, 0, len(ctl.inputs))
+	for _, v := range ctl.inputs {
+		res = append(res, v)
+	}
+	return res
+}
+
+func (ctl *InputsController) Reset(inputs []*wrapinput.Model) {
+	for k := range ctl.inputs {
+		ctl.inputs[k].ResetValidation()
+	}
+}
+
+func (ctl *InputsController) Validate(inputs []*wrapinput.Model) bool {
+	hasErrors := false
+	for k, input := range ctl.inputs {
+		setError := func(err error) bool {
+			if err != nil {
+				input.SetError(err)
+				debug.Output(err.Error())
+				hasErrors = true
+				return true
+			}
+			return false
+		}
+
+		if input.Required && len(input.Value()) == 0 {
+			setError(errors.New("required"))
+			continue
+		}
+
+		switch k {
+		case keyTowerVersion:
+			version, err := gcp.GetTowerImageVersion(input.Value())
+			if setError(err) {
+				debug.Output("convert tower version got error: %v", err)
+				break
+			}
+
+			input.SetConverted(version)
+			debug.Output("convert tower version is %s", version)
+
+		case keyVirtualMachine:
+			projectInput := ctl.inputs[keyProject]
+			var err error
+			ctl.args[keyVirtualMachineIP], err = ctl.getVirtualMachineIP(projectInput.Value(), ctl.inputs[keyZone].Value(), input.Value())
+			if setError(err) {
+				break
+			}
+
+			input.SetInfo(fmt.Sprintf("found VM with IP %s", ctl.args[keyVirtualMachineIP]))
+
+		}
+	}
+
+	return !hasErrors
+}
+
+func (ctl InputsController) GetRunController() controller.Run {
+	for _, k := range ctl.inputKeys {
+		ctl.args[k] = ctl.inputs[k].Value()
+	}
+
+	return NewRunController(ctl.args)
+}
+
+func (ctl InputsController) GetName() string {
+	return "Google Cloud Platform Upgrade"
+}
+
+func (ctl *InputsController) GetInputs() []*wrapinput.Model {
+	ctl.inputKeys = initialInputKeys
+
+	inputs := make([]*wrapinput.Model, 0, len(ctl.inputKeys))
+	for _, k := range ctl.inputKeys {
+		inputs = append(inputs, ctl.inputs[k])
+	}
+	return inputs
+}
+
+func (ctl *InputsController) RecreateCommand() string {
+	for _, key := range ctl.inputKeys {
+		ctl.args[key] = ctl.inputs[key].Value()
+	}
+	return recreateCommand(ctl.args)
+}
+
+func contains(m map[string]string, key string) bool {
+	_, ok := m[key]
+	return ok
+}
+
+func (ctl *InputsController) getVirtualMachineIP(projectID string, zone string, vmName string) (string, error) {
```

Comment:
Just a sanity check - does this work when the VM is stopped?

Reply:
There's no IP when the VM is stopped, but it does manage to find the VM -- updated the control flow for this.